packages/graalvm-kt/src/main/kotlin/elide/runtime/plugins/kotlin/Kotlin.kt

```diff
@@ -14,50 +14,52 @@
 import elide.runtime.core.EnginePlugin.Key
-import elide.runtime.core.PolyglotContext
-import elide.runtime.core.PolyglotContextBuilder
 import elide.runtime.core.getOrInstall
 import elide.runtime.plugins.AbstractLanguagePlugin
-import elide.runtime.plugins.AbstractLanguagePlugin.LanguagePluginManifest
 import elide.runtime.plugins.jvm.Jvm
+import elide.runtime.plugins.kotlin.shell.GuestKotlinInterpreter
 
-@Suppress("unused") @DelicateElideApi public class Kotlin private constructor(
-  private val config: KotlinConfig,
-  private val resources: LanguagePluginManifest,
-) {
-  private fun initializeContext(context: PolyglotContext) {
-    // apply init-time settings
-    config.applyTo(context)
-
-    // run embedded initialization code
-    initializeEmbeddedScripts(context, resources)
-  }
-  
-  private fun configureContext(builder: PolyglotContextBuilder) {
-    // nothing to do here
-  }
-  
+/**
+ * Runtime plugin adding support for evaluating Kotlin in an interactive shell. Applying this plugin will automatically
+ * install the [Jvm] plugin.
+ *
+ * This plugin adds custom guest classpath entries using the [Jvm] plugin configuration which are necessary for the
+ * [GuestKotlinInterpreter] to function, such as the Kotlin standard library, the scripting runtime, and a custom
+ * JAR providing helper classes used to interface with the guest context from the host.
+ *
+ * @see [GuestKotlinInterpreter]
+ */
+@DelicateElideApi public class Kotlin private constructor(public val config: KotlinConfig) {
   public companion object Plugin : AbstractLanguagePlugin<KotlinConfig, Kotlin>() {
+    private const val GUEST_CLASSPATH_KEY = "classpath"
+
     private const val KOTLIN_LANGUAGE_ID = "kt"
     private const val KOTLIN_PLUGIN_ID = "Kotlin"
-    
+
     override val languageId: String = KOTLIN_LANGUAGE_ID
     override val key: Key<Kotlin> = Key(KOTLIN_PLUGIN_ID)
-    
+
+    private fun resolveGuestClasspathEntries(manifest: LanguagePluginManifest): List<String> {
+      return manifest.resources[GUEST_CLASSPATH_KEY]?.map {
+        Kotlin::class.java.getResource("${manifest.root}/$it")?.toString() ?: error(
+          "Failed to resolve embedded classpath element: $it",
+        )
+      } ?: emptyList()
+    }
+
     override fun install(scope: InstallationScope, configuration: KotlinConfig.() -> Unit): Kotlin {
-      // apply the JVM plugin first
-      scope.configuration.getOrInstall(Jvm)
-      
-      // apply the configuration and create the plugin instance
-      val config = KotlinConfig().apply(configuration)
       val resources = resolveEmbeddedManifest(scope)
-      val instance = Kotlin(config, resources)
 
-      // subscribe to lifecycle events
-      scope.lifecycle.on(ContextCreated, instance::configureContext)
+      // apply the JVM plugin first, and register the custom classpath entries
+      scope.configuration.getOrInstall(Jvm).config.apply {
+        classpath(resolveGuestClasspathEntries(resources))
+      }
```

Comment:
### Guest classpath entries

It's tempting to package all the guest dependencies into a tarball and serve it using the VFS, but Espresso does not yet support virtualized I/O!. This means (for now at least) we need to keep the JARs that go into the guest classpath in our embedded resources, and serve them directly from there. Fortunately Espresso is able to load them from there, though I'm not sure how (or even if) this will work in native mode.